src/tls_openssl.c

```diff
@@ -1032,6 +1058,7 @@ _tls_cert_read_p12(xmpp_conn_t *conn, EVP_PKEY **pkey, STACK_OF(X509) * *ca)
 
 static X509 *_tls_cert_read(xmpp_conn_t *conn)
 {
+    check_providers(conn->ctx);
```

Comment:
Since this just prints an error, it can be done once in tls_initialize().

Reply:
better now? :)